File: Connectors/InteractiveBrokers/Java/Client/ExecutionFilter.java
/* Copyright (C) 2013 Interactive Brokers LLC. All rights reserved.  This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

package com.ib.client;

public class ExecutionFilter{
    public int 		m_clientId; // zero means no filtering on this field
    public String 	m_acctCode;
    public String 	m_time;
    public String 	m_symbol;
    public String 	m_secType;
    public String 	m_exchange;
    public String 	m_side;

    public ExecutionFilter() {
        m_clientId = 0;
    }

    public ExecutionFilter( int p_clientId, String p_acctCode, String p_time,
    		String p_symbol, String p_secType, String p_exchange, String p_side) {
        m_clientId = p_clientId;
        m_acctCode = p_acctCode;
        m_time = p_time;
        m_symbol = p_symbol;
        m_secType = p_secType;
        m_exchange = p_exchange;
        m_side = p_side;
    }

    public boolean equals(Object p_other) {
        boolean l_bRetVal = false;

        if ( p_other == null ) {
            l_bRetVal = false;
		}
        else if ( this == p_other ) {
            l_bRetVal = true;
        }
        else {
            ExecutionFilter l_theOther = (ExecutionFilter)p_other;
            l_bRetVal = (m_clientId == l_theOther.m_clientId &&
                    m_acctCode.equalsIgnoreCase( l_theOther.m_acctCode) &&
                    m_time.equalsIgnoreCase( l_theOther.m_time) &&
                    m_symbol.equalsIgnoreCase( l_theOther.m_symbol) &&
                    m_secType.equalsIgnoreCase( l_theOther.m_secType) &&
                    m_exchange.equalsIgnoreCase( l_theOther.m_exchange) &&
                    m_side.equalsIgnoreCase( l_theOther.m_side) );
        }
        return l_bRetVal;
    }
}

File: Connectors/InteractiveBrokers/CSharp/ExecutionFilter.cs
/* Copyright (C) 2013 Interactive Brokers LLC. All rights reserved.  This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;

namespace IBApi
{
    /**
     * @class ExecutionFilter
     * @brief when requesting executions, a filter can be specified to receive only a subset of them
     * @sa Contract, Execution, CommissionReport
     */
    public class ExecutionFilter
    {
        
        private int clientId;
        private string acctCode;
        private string time;
        private string symbol;
        private string secType;
        private string exchange;
        private string side;

        /**
         * @brief The API client which placed the order
         */
        public int ClientId
        {
            get { return clientId; }
            set { clientId = value; }
        }

        /**
        * @brief The account to which the order was allocated to
        */
        public string AcctCode
        {
            get { return acctCode; }
            set { acctCode = value; }
        }

        /**
         * @brief Time from which the executions will be brough yyyymmdd hh:mm:ss
         * Only those executions reported after the specified time will be returned.
         */
        public string Time
        {
            get { return time; }
            set { time = value; }
        }

        /**
        * @brief The instrument's symbol
        */
        public string Symbol
        {
            get { return symbol; }
            set { symbol = value; }
        }

        /**
         * @brief The Contract's security's type (i.e. STK, OPT...)
         */
        public string SecType
        {
            get { return secType; }
            set { secType = value; }
        }

        /**
         * @brief The exchange at which the execution was produced
         */
        public string Exchange
        {
            get { return exchange; }
            set { exchange = value; }
        }

        /**
        * @brief The Contract's side (Put or Call).
        */
        public string Side
        {
            get { return side; }
            set { side = value; }
        }

        public ExecutionFilter()
        {
            clientId = 0;
        }

        public ExecutionFilter(int clientId, String acctCode, String time,
                String symbol, String secType, String exchange, String side)
        {
            ClientId = clientId;
            AcctCode = acctCode;
            Time = time;
            Symbol = symbol;
            SecType = secType;
            Exchange = exchange;
            Side = side;
        }

        public override bool Equals(Object other)
        {
            bool l_bRetVal = false;

            if (other == null)
            {
                l_bRetVal = false;
            }
            else if (this == other)
            {
                l_bRetVal = true;
            }
            else
            {
                ExecutionFilter l_theOther = (ExecutionFilter)other;
                l_bRetVal = (clientId == l_theOther.clientId &&
                    String.Compare(AcctCode, l_theOther.acctCode, true) == 0 &&
                    String.Compare(time, l_theOther.Time, true) == 0 &&
                    String.Compare(symbol, l_theOther.Symbol, true) == 0 &&
                    String.Compare(secType, l_theOther.SecType, true) == 0 &&
                    String.Compare(exchange, l_theOther.Exchange, true) == 0 &&
                    String.Compare(side, l_theOther.Side, true) == 0);
            }
            return l_bRetVal;
        }
    }
}
